wrangler-core/src/main/java/co/cask/directives/date/FormatDate.java

```diff
@@ -76,21 +78,32 @@ public void destroy() {
     for (Row row : rows) {
       Row dt = new Row(row);
       int idx = dt.find(column);
-      if (idx != -1) {
-        Object object = row.getValue(idx);
-        if (object != null && object instanceof Date) {
-          dt.setValue(idx, destinationFmt.format((Date) object));
-        } else {
+
+      if (idx == -1) {
+        throw new DirectiveExecutionException(toString() + " : '" + column + "' column is not defined in the row. " +
+                                                "Please check the wrangling step."
+        );
+      }
+
+      Object object = row.getValue(idx);
+
+      if (object != null) {
+        try {
+          if (object instanceof LocalDate) {
```

Comment:
when is it LocalDate? I thought we were using ZonedDateTime everywhere?

Reply:
`FormatDate` can be applied on data read directly from connections like sql database, bigquery, spanner etc. If user wants to format `LocalDate` or `LocalTime` read from the connection, he/she can still do it using this directive.

Reply:
It doesn't matter where the data came from. 

LocalTime doesn't make any sense, it's not tied to any day of the year. ZonedDateTime (timestamp logical type) should be supported.